### PR TITLE
Storage: list filtering and root folder support 

### PR DESCRIPTION
**Pull request by ArturWierzbicki**
extracted from https://github.com/grafana/grafana/pull/46363/files

### CODE REVIEW

File: pkg/infra/filestorage/filestorage.go [hunk 1]
```diff
@@ -21,140 +12,6 @@ const (
...
-}
-
-func (b service) IsFolderEmpty(ctx context.Context, path string) (bool, error) {
-	return true, errors.New("not implemented")
-}
-
-func (b service) close() error {
-	return b.grafanaDsStorage.close()
+	// TODO: initialize deps if needed for s3/gcs
+	return &dummyFileStorage{}, nil
```
Comment (ryantxu) [inline pkg/infra/filestorage/filestorage.go:16]: :thinking: it seems we do not really need a service if we shift to `storage` as a service

can each implementation do its own initialization?

Comment (ryantxu) [reply pkg/infra/filestorage/filestorage.go]: similar question... do we need the feature flag anymore?

Comment (ArturWierzbicki) [reply pkg/infra/filestorage/filestorage.go]: nope! nor do we need the dummy implementation, both gone

**Review by ryantxu — APPROVED** [review]
Looks good! fewer services is better :)

**Pull request merged**